### PR TITLE
rewrite tunnel.sh to also work on macOS

### PR DESCRIPTION
**Pull request by pkyeck**
Hi + thanks for your repository.

I'm no bash expert but I tried to make the SSH tunnel also work on macOS without breaking your linux approach.
Couldn't check if it's working as expected under linux ... maybe you could.

### CODE REVIEW

File: tunnel.sh [hunk 1]
```diff
@@ -8,18 +7,22 @@ GATEWAY="$6"
 SHELL="$7"
 MPID="$8"
 
+ABSPATH=$(cd "$(dirname "$0")"; pwd -P)
```
Comment (pkyeck) [inline tunnel.sh:10]: I had spaces in my folder-names and that's why your `$PWD/$0` didn't work for me

File: tunnel.sh [hunk 2]
```diff
@@ -8,18 +7,22 @@ GATEWAY="$6"
...
+
 if [ -z "$MPID" ] ; then
   echo '{}'
-  p=`ps --pid $PPID -o ppid --no-headers`
-  nohup timeout $TIMEOUT bash $PWD/$0 $@ $p <&- >&- 2>&- &
+  p=`ps -p $PPID -o "ppid="`
```
Comment (pkyeck) [inline tunnel.sh:14]: macOS' `ps` doesn't have `--pid` but instead `-p`. `--no-headers` isn't there as well but we can achieve the same using `-o "ppid="`

File: tunnel.sh [hunk 3]
```diff
@@ -8,18 +7,22 @@ GATEWAY="$6"
...
+  nohup timeout $TIMEOUT bash "$ABSPATH/tunnel.sh" $@ $p <&- >&- 2>&- &
   exit 0
 fi
 
 $SSH -N -L $LOCAL_PORT:$TARGET_HOST:$TARGET_PORT $GATEWAY &
 CPID=$!
 
 while true ; do
-  [ -d /proc/$MPID ] || break
```
Comment (pkyeck) [inline tunnel.sh:22]: and there is no `/proc/<PID>` under macOS – so I check if `ps -p $MPID` succeeds or fails

**Comment by pkyeck** [comment]
I made some comments to explain the changes here: https://github.com/flaupretre/terraform-ssh-tunnel/pull/2/files

**Comment by flaupretre** [comment]
Thanks for this PR. I judt tried it on Linux and it seems to work as usual.

**Pull request merged**